[HCCDOC-1094] Insights DataGather (#63976)
openshift/openshift-docs: modules/running-insights-operator-gather.adoc

:FeatureName: The `InsightsDataGather` custom resource 
include::snippets/technology-preview.adoc[] 

[NOTE]
====
Enabling Technology Preview in your cluster will cause Insights Operator to run gather operations in individual pods. This is part of the Technology Preview feature set for the Insights Operator and supports the new data gathering features.
====

.Prerequisites

* You are logged in to the {product-title} web console as a user with `cluster-admin` role.

.Procedure

. Navigate to *Administration* -> *CustomResourceDefinitions*.
. On the *CustomResourceDefinitions* page, use the *Search by name* field to find the *DataGather* resource definition and click it.
. On the *CustomResourceDefinition details* page, click the *Instances* tab.
. Click *Create DataGather*.
. To create a new `DataGather` operation, edit the configuration file:
+
[source,yaml]
----
apiVersion: insights.openshift.io/v1alpha1
kind: DataGather
metadata:
  name: <your_data_gather> <1>
spec:
 gatherers: <2>
   - name: workloads 
     state: Disabled
----
+
--

> <1> Replace the `<your_data_gather>` with a unique name for your gather operation. 

Reviewer: Perhaps I would also add the info that it must not be prefixed with `periodic-gathering`

Author: Done

<2> Enter individual gather operations to disable the under the `gatherers` parameter. This example disables the `workloads` data gather. To run the default list of gather operations, leave the `spec` parameter empty.
--
+
. Click *Save*.
. Navigate to *Workloads* -> *Pods*.
. On the Pods page, select All Projects from the Project list, and then set Show default projects to on.

> . Select the `openshift-config` project from the Projects list. 

Reviewer: same here as below please. `openshift-insights` namespace.

Author: good catch!

+

> You will see your new gather operation under the list of pods in the `openshift-config` project. 

Reviewer: `openshift-insights` namespace here.

Author: good catch!

> You will see your new gather operation under the list of pods in the `openshift-insights` project. 

Reviewer 2: Here we can specify that the newly created gather operation will be prefixed with the given data gather instance name (for easiness on distinguishing between several gather operations that can happen simultaneously)

// Module included in the following assemblies:
//
// * support/remote_health_monitoring/using-insights-operator.adoc

:_content-type: PROCEDURE
[id="running-insights-operator-gather_{context}"]
= Running an Insights Operator gather operation

You can run Insights Operator data gather operations on-demand. The following procedure describes how to run the default list of gather operations excluding the `workloads` operation. You can customize the on-demand gather function to exclude any gather operations you choose. If you have previously disabled Insights Operator gather operations in your cluster, this procedure will override those parameters.  

:FeatureName: The `DataGather` custom resource 
include::snippets/technology-preview.adoc[] 

[NOTE]
====
If you enable Technology Preview in your cluster, Insights Operator runs gather operations in individual pods. This is part of the Technology Preview feature set for the Insights Operator and supports the new data gathering features.
====

> .Prerequisites

Reviewer 3: Should these prereqs include "You have the oc CLI installed on your cluster" for the CLI method?

* You have access to the cluster as a user with the `cluster-admin` role.

> .Procedure

Reviewer 3: Honestly, considering that there's no overlap between the CLI and web console methods, this should really be two separate procedures. You could probably keep them both in this module but have two level-two headings like such:

`== Running an Insights Operator gather operation using the web console`

`== Running an Insights Operator gather operation using the CLI`

. To run the gather operation from the web console:
.. Navigate to *Administration* -> *CustomResourceDefinitions*.
.. On the *CustomResourceDefinitions* page, use the *Search by name* field to find the *DataGather* resource definition and click it.
.. On the *CustomResourceDefinition details* page, click the *Instances* tab.
.. Click *Create DataGather*.
.. To create a new `DataGather` operation, edit the configuration file:
+
[source,yaml]
----
apiVersion: insights.openshift.io/v1alpha1
kind: DataGather
metadata:
  name: <your_data_gather> <1>
spec:
 gatherers: <2>
   - name: workloads 
     state: Disabled
----
+
--
<1> Replace the `<your_data_gather>` with a unique name for your gather operation.

> <2> Enter individual gather operations to disable the under the `gatherers` parameter. This example disables the `workloads` data gather operation and will run the remainder of the default operations. To run the complete list of default gather operations, leave the `spec` parameter empty. You can find the complete list of gather operations in the Insights Operator documentation. 

Reviewer 3: I think there's a word missing here: "to disable the <something> under the `gatherers` parameter."

--
+
.. Click *Save*.
.. Navigate to *Workloads* -> *Pods*.

> .. On the Pods page, select All Projects from the Project list, and then set Show default projects to on.

Reviewer 3: Suggestion: Either put "on" in quotation marks or reword this step:

```suggestion
.. On the Pods page, select All Projects from the Project list, and then turn on Show default projects.
```

.. Select the `openshift-insights` project from the Projects list. 
+
You will see your new gather operation prefixed with your chosen name under the list of pods in the `openshift-insights` project. Upon completion, the Insights Operator automatically uploads the data to Red Hat for processing. 
+

> . To run the gather operation from the CLI, Enter the following command: 

Reviewer 3: "Enter" should be lowercase:

```suggestion
. To run the gather operation from the CLI, enter the following command: 
```

+
[source,terminal]
----

> oc apply -f <your-datagather-definition>.yaml 

Reviewer 3: I think this needs a $ in front of the command:

```suggestion
$ oc apply -f <your-datagather-definition>.yaml 
```

// Module included in the following assemblies:
//
// * support/remote_health_monitoring/using-insights-operator.adoc

:_content-type: PROCEDURE
[id="running-insights-operator-gather_{context}"]
= Running an Insights Operator gather operation

> You can run Insights Operator data gather operations on demand. The following procedure describes how to run the default list of gather operations excluding the `workloads` operation. You can customize the on demand gather function to exclude any gather operations you choose. If you have previously disabled Insights Operator gather operations in your cluster, this procedure will override those parameters.  

Reviewer: > You can customize the on demand gather function to exclude any gather operations you choose

That's true, but this should go with some bold warning that if you exclude something then your active Insights recommendations may disappear, which might be an unwanted effect. In other words, you should know what you're doing.

* You are logged in to the {product-title} web console as a user with the `cluster-admin` role.

.Procedure

. Navigate to *Administration* -> *CustomResourceDefinitions*.
. On the *CustomResourceDefinitions* page, use the *Search by name* field to find the *DataGather* resource definition and click it.
. On the *CustomResourceDefinition details* page, click the *Instances* tab.
. Click *Create DataGather*.
. To create a new `DataGather` operation, edit the configuration file:
+
[source,yaml]
----
apiVersion: insights.openshift.io/v1alpha1
kind: DataGather
metadata:
  name: <your_data_gather> <1>
spec:
 gatherers: <2>
   - name: workloads 
     state: Disabled
----
+
--
<1> Replace the `<your_data_gather>` with a unique name for your gather operation.
<2> Enter individual gather operations to disable under the `gatherers` parameter. This example disables the `workloads` data gather operation and will run the remainder of the default operations. To run the complete list of default gather operations, leave the `spec` parameter empty. You can find the complete list of gather operations in the Insights Operator documentation. 
--
+
. Click *Save*.
. Navigate to *Workloads* -> *Pods*.

> . On the Pods page, select All Projects from the Project list, and then turn on Show default projects.

Reviewer 4: ```suggestion
. On the *Pods* page, select *All Projects* from the *Project* list, and then turn on *Show default projects*.
```
Also: is it "Project" or "Projects"?

> . Select the `openshift-insights` project from the Projects list. 

Reviewer 4: ```suggestion
. Select the `openshift-insights` project from the *Projects* list. 
```

Author: fixed

+

> You will see your new gather operation prefixed with your chosen name under the list of pods in the `openshift-insights` project. Upon completion, the Insights Operator automatically uploads the data to Red Hat for processing. 

Reviewer 4: Same comments here as the same spot from the CLI procedure.

Author: Added verification steps

[id="running-insights-operator-gather-openshift-cli"]
== Running an Insights Operator gather operation using the OpenShift CLI

.Prerequisites

* You are logged in to {product-title} as a user with the `cluster-admin` role.

.Procedure

> . Enter the following command to run the gather operation: 

Author: fixed

+
[source,terminal]
----
$ oc apply -f <your-datagather-definition>.yaml 
----
+

> Replace `<your-datagather-definition>.yaml` with a configuration file using the same parameters as the above procedure.

Author: I just added the whole configuration as we are going modular on this one and I guess it could be used without the web console version at some point.

+

> You will see your new gather operation prefixed with your chosen name under the list of pods in the `openshift-insights` project. Upon completion, the Insights Operator automatically uploads the data to Red Hat for processing.

Author: I added the navigation to verification steps as well
